### PR TITLE
S10 refactoring

### PR DESCRIPTION
**Pull request by lingsat**

### CODE REVIEW

File: src/constants.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const debounceDelay = 500;
```
Comment (Yuliia143) [inline src/constants.ts]: should be uppercase

File: src/features/products/components/Filter/Filter.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        onClick={toggleFilter}></div>
+      <aside
+        className={`filter ${showFilter ? "filter--show" : ""}`}
+        onClick={handleFilterPropagation}>
+        <div className="filter__block">
+          <h3 className="filter__title">Categories</h3>
+          <ul className="filter-categories">
+            <li
+              className={`filter-categories__item ${
+                filter.category === "" ? "filter-categories__item--active" : ""
```
Comment (Yuliia143) [inline src/features/products/components/Filter/Filter.tsx]: Simplify condition

File: src/features/products/components/Filter/Filter.tsx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        <div className="filter__block">
+          <h3 className="filter__title">Rating</h3>
+          <ul className="filter-rating">
+            {starsArr.map((starCount, index) => (
+              <li key={`filterstars-${starCount}-${index}`}>
+                <label className="filter__label">
+                  <input
+                    className="filter__input"
+                    type="checkbox"
+                    checked={filter.stars.includes(+starCount)}
```
Comment (Yuliia143) [inline src/features/products/components/Filter/Filter.tsx]: logic in jsx

File: src/features/products/components/Navigation/Navigation.tsx [hunk 4]
```diff
@@ -1,8 +1,9 @@
 import React, { FC } from "react";
 import { Link } from "react-router-dom";
-import "./ListNavigation.scss";
 
-const ListNavigation: FC = () => {
+import "./Navigation.scss";
+
+const Navigation: FC = () => {
   return (
     <ul className="list-nav">
```
Comment (Yuliia143) [inline src/features/products/components/Navigation/Navigation.tsx]: Simplify

File: src/features/products/pages/Home/Home.tsx [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import React, { FC } from "react";
 import { Link } from "react-router-dom";
 
-const HomePage: FC = () => {
+const Home: FC = () => {
   return (
```
Comment (Yuliia143) [inline src/features/products/pages/Home/Home.tsx]: x2

File: src/features/products/pages/ProductsList/ProductsList.scss [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+  }
+}
+
+@media (max-width: 575.98px) {
+  .products-list__title {
+    font-size: 1.5rem;
+  }
+
+  .products-list__header {
+    margin-bottom: 0;
```
Comment (Yuliia143) [inline src/features/products/pages/ProductsList/ProductsList.scss]: Coud be : 
.products-list {
    &&__title {}
    &&_header {}
}

File: src/constants.ts (hunk 1 above)
Comment (lingsat) [reply src/constants.ts]: Fixed - DEBOUNCE_DELAY

File: src/features/products/components/Filter/Filter.tsx (hunk 2 above)
Comment (lingsat) [reply src/features/products/components/Filter/Filter.tsx]: Fixed -  filter.category === "" replace with !filter.category

File: src/features/products/components/Filter/Filter.tsx (hunk 3 above)
Comment (lingsat) [reply src/features/products/components/Filter/Filter.tsx]: Fixed - create universal function that checks if item includes in array

File: src/features/products/components/Navigation/Navigation.tsx (hunk 4 above)
Comment (lingsat) [reply src/features/products/components/Navigation/Navigation.tsx]: Remove FC from all components without props

File: src/features/products/pages/Home/Home.tsx (hunk 5 above)
Comment (lingsat) [reply src/features/products/pages/Home/Home.tsx]: Remove FC from all components without props

File: src/features/products/pages/ProductsList/ProductsList.scss (hunk 6 above)
Comment (lingsat) [reply src/features/products/pages/ProductsList/ProductsList.scss]: Fixed

File: src/features/products/components/Navigation/Navigation.tsx (hunk 4 above)
Comment (Yuliia143) [reply src/features/products/components/Navigation/Navigation.tsx]: I meant to remove the unnecessary return

Comment (lingsat) [reply src/features/products/components/Navigation/Navigation.tsx]: Removed

**Review by Yuliia143 — APPROVED** [review]

**Pull request merged**